fl4health/preprocessing/autoencoders/dim_reduction.py

```diff
@@ -0,0 +1,84 @@
+from abc import ABC
+from pathlib import Path
+
+import torch
+
+
+class Processing(ABC):
+    """
+    Abstact class to add processing methods.
+    """
+
+    def __init__(
+        self,
+        checkpointing_path: Path,
+    ) -> None:
+        self.checkpointing_path = checkpointing_path
+        self.device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+        self.load_autoencoder()
+
+    def load_autoencoder(self) -> None:
+        autoencoder = torch.load(self.checkpointing_path)
+        autoencoder.eval()
+        self.autoencoder = autoencoder.to(self.device)
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}"
+
+
+class AeProcessor(Processing):
+    """Transformer processor to encode the data using basic autoencoder."""
+
+    def __init__(
```

Comment:
If you're not doing anything other than calling the super init, I believe you can omit it from the class.